flink-streaming-java/src/main/java/org/apache/flink/streaming/api/connector/sink2/StandardSinkTopologies.java

```diff
@@ -34,7 +41,21 @@ private StandardSinkTopologies() {}
      */
     public static <CommT> void addGlobalCommitter(
             DataStream<CommittableMessage<CommT>> committables,
-            SerializableSupplier<Committer<CommT>> globalCommitterFactory) {
-        // TODO: FLINK-25726
+            SerializableSupplier<Committer<CommT>> globalCommitterFactory,
+            SerializableSupplier<SimpleVersionedSerializer<CommT>> committableSerializer) {
+        final PhysicalTransformation<Void> transformation =
+                (PhysicalTransformation<Void>)
+                        committables
+                                .global()
+                                .transform(
+                                        GLOBAL_COMMITTER_TRANSFORMATION_NAME,
+                                        Types.VOID,
+                                        new GlobalCommitter<>(
+                                                globalCommitterFactory, committableSerializer))
+                                .getTransformation();
+        transformation.setChainingStrategy(ChainingStrategy.ALWAYS);
```

Comment:
since the parallelism will be set to be 1, should we use `ChainingStrategy.NEVER` to improve the resource guarantee?

Reply:
Actually, there have been preexisting tests to verify that the chaining of the `GlobalCommitter` is `ALWAYS` I did not want to change that behavior.